docs/spec/lightclient/verification.md

```diff
@@ -0,0 +1,925 @@
+
+> Note that for liveness, initially having a *trustedHeader* within
+> the *trustinPeriod* is not sufficient. However, as this
+> specification will leave some freedom with respect to the strategy
+> in which order to download intermediate headers, we do not give a
+> more precise liveness specification here. After giving the
+> specification of the protocol, we will discuss some liveness
+> scenarios [below](#liveness-scenarios).
+
+### Solving the sequential specification
+
+This specification provides a partial solution to the sequential specification.
+The *Verifier* solves the invariant of the sequential part
+
+[**[LCV-DIST-SAFE]**](#lcv-vc-inv) => [**[LCV-SEQ-SAFE]**](#lcv-seq-inv)
+
+In the case the primary is correct, and there is a recent header in *LightStore*, the verifier satisfies the liveness requirements.
+
+/\ "correct primary"  
+/\ always \E verified header in LightStore. header.Time > now - *trustingPeriod*  
+/\ [**[LCV-A-Comm]**](#lcv-a-comm) /\
+       [**[TMBC-CorrFull]**][TMBC-CorrFull-link] /\
+       [**[LCV-DIST-LIVE]**](#lcv-vc-live)  
+       => [**[LCV-SEQ-LIVE]**](#lcv-seq-live)
+
+
+# Part IV - Light Client Verification Protocol
+
+We provide a specification for Light Client Verification. The local
+code for verification is presented by a sequential function
+`VerifyToTarget` to highlight the control flow of this functionality.
+We note that if a different concurrency model is considered for
+an implementation, the sequential flow of the function may be
+implemented with mutexes, etc. However, the light client verification
+is partitioned into three blocks that can be implemented and tested
+independently:
+
+- `FetchLightBlock` is called to download a light block (header) of a
+  given height from a peer.
+- `ValidAndVerified` is a local code that checks the header.
+- `Pivot` decides which height to try to verify next. We keep this
+  underspecified as different implementations (currently in Goland and
+  Rust) may implement different optimizations here. We just provide
+  necessary conditions on how the height may evolve.
+  
+
+> `ValidAndVerified` is the function that is sometimes called "Light
+> Client" in the IBC context.
+
+
+## Definitions
+
+### Data Types
+
+The core data structure of the protocol is the LightBlock.
+
+```go
+type LightBlock struct {
+		Header          Header
+		Commit          Commit
+		Validators      ValidatorSet
+		NextValidators  ValidatorSet
+		Provider        PeerID
+}
+```	
+
+#### **[LCV-DATA-LIGHTSTORE]**:
+
+
+LightBlocks are stored in a structure which stores all LightBlock from
+initialization or received from peers.
+
+```go
+type LightStore struct {
+	...
+}
+
+```
+
+
+Each LightBlock is in one of the following states:
+```go
+type VerifiedState int
+
+const (
+	StateUnverified = iota + 1
+	StateVerified
+	StateFailed
+	StateTrusted
+)
+```
+
+> Only the detector module sets a lightBlock state to `StateTrusted`
```

Comment:
It is not clear what is the purpose of this `Trusted` state, since it does not appear anywhere in this spec except in this definition, nor in the verification spec.

Reply:
Edit: except in this definition and the postcondition of `LatestVerified`

Reply:
The idea was that `StateTrusted` is the state after detection was passed. Since we define the data structure I have put it here. But I forgot to set the state at the end of detection. I will add that now! Thanks!!